sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -118,8 +118,8 @@ private[hive] object HiveShim {
    *
    * @param functionClassName UDF class name
    */
-  private[hive] case class HiveFunctionWrapper(var functionClassName: String)
-    extends java.io.Externalizable {
+  private[hive] case class HiveFunctionWrapper(var functionClassName: String,
+    var instance: AnyRef = null) extends java.io.Externalizable with Logging {
```

Comment:
For "instance", just conformed with "functionClassName", which is public. For "Logging", I've used it in debugging and forgot to remove. I'll addressed them.